File: sujeong/week3/기능개발.swift
import Foundation

func solution(_ progresses:[Int], _ speeds:[Int]) -> [Int] {
    var answer:[Int] = []
    var datesNeededList:[Int] = []
    
    for (index,progress) in progresses.enumerated() {
        let datesNeeded = Int(ceil((Double(100-progress)/Double(speeds[index]))))
        datesNeededList.append(datesNeeded)
    }
    
    var currentReleaseDate:Int = datesNeededList[0]
    var numOfRelease:Int = 0
    for datesNeeded in datesNeededList {
        if datesNeeded <= currentReleaseDate {
            numOfRelease += 1
        } else {
            answer.append(numOfRelease)
            numOfRelease = 1
            currentReleaseDate = datesNeeded
        }
    }
    answer.append(numOfRelease)

    return answer
}

File: hyein/week3/기능개발.py
from collections import deque

def solution(progresses, speeds):
    answer = []
    progresses = deque(progresses)
    speeds = deque(speeds)
    
    count = 0
    days = 0

    while progresses:
        if progresses[0] + speeds[0] * days >= 100:
            count += 1
            progresses.popleft()
            speeds.popleft()
        else:
            if count > 0:
                answer.append(count)
                count = 0 
                
            days += 1    
    
    answer.append(count)
    return answer